### DISCUSSION TITLE
Keeping focused non-modal Dialog above others with `z-index`

### DISCUSSION THREAD

**Discussion by ignas-k** [iTwin/iTwinUI#2104] [Ideas]
In a scenario where multiple non-modal dialogs are open, it would be nice if focusing on a single dialog would bring that dialog to the top (through z-index). Would it be possible for this to be handled by iTwinUI? Example:

![msedge_LHAPSmKQSS](https://github.com/iTwin/iTwinUI/assets/138560995/065e7a48-e5b2-4889-925d-fa07afa006c2)

**Comment by Ben-Pusey-Bentley** [comment]
This should be possible for iTwinUI to handle, since it seems to just be some styling. Some additional information on how exactly you are implementing this would be nice. Are you using the `:focus` pseudo-class to set the z-index for the focused dialog, or are you using some other method?

**Reply by ignas-k** [reply]
Our current implementation is passing `style` prop with `zIndex` to `Dialog` and `onPointerDown` prop to `Dialog.Main` so something like this:

```      
     <Dialog style={style} {...props}>
        <Dialog.Main
          onPointerDown={handleContainerPointerDown}
          {...mainProps}
        >
          <Dialog.Content>{content}</Dialog.Content>
        </Dialog.Main>
      </Dialog>
```

Not sure if this is the best way to implement this functionality though.

**Reply by mayank99** [reply]
`onPointerDown` will only cover pointers. We need to also cover other input devices, probably using `onFocus`.
